### DISCUSSION TITLE
Facial recognition for Frigate

### DISCUSSION THREAD

**Discussion by DrSpaldo** [blakeblackshear/frigate#6735] [Ideas]
@blakeblackshear @NickM-27 I am not sure if Frigate has had any consideration into implementing facial recognition into the NVR itself or not. I am also not sure if many here are following the development of the [Immich](https://github.com/immich-app/immich) photo & video self-hosted app. It is essentially a replacement for Google Photos that is hosted locally. They have recently implemented the long awaited [facial recognition feature](https://github.com/immich-app/immich/pull/2180)

It is worth taking a look at. I have tried every self hosted photo server/software that I could get my hands on, some have a nice interface, but none have the same level of facial recognition that Immich has managed to get going. 

I can post this into [Double Take](https://github.com/jakowenko/double-take) that @jakowenko has made, but I just wanted to touch base here first...

**Comment by NickM-27** [comment]
I've found compreface to work quite well. https://github.com/blakeblackshear/frigate/issues/260 is also pinned

**Reply by DrSpaldo** [reply]
Thanks, I will have a read through [260](https://github.com/blakeblackshear/frigate/issues/260). When I initially tried Compreface with all three running, I didn't have much luck on my cameras..

**Reply by NickM-27** [reply]
which model were you running? compreface has multiple

**Reply by DrSpaldo** [reply]
I'll give it another go and use arcface-r100-gpu this time. If you are interested though, [Immich ](https://github.com/immich-app/immich)use [insightface](https://github.com/deepinsight/insightface)

**Reply by NickM-27** [reply]
I've found great success with arcface-r100 and being meticulous about which images are used for training. I did a lot in the beginning of training with 2 images and then refreshing the `matches` to see if the scores went up or down. Did this many iterations until it was reliable and it has remained reliable even after moving to a new house

**Reply by DrSpaldo** [reply]
I am sure it isn't the one I used originally, although it is a while ago... So I am happy to give it a go. When you say you used 2 images for training, how exactly do you mean? 

It is really strange, when I run Double Take as an addon within home assistant, it connects to Frigate fine. But when I run it as another container in Unraid it doesn't connect. The strange thing is that Frigate is in it's own container, with the same IP, so it shouldn't make any difference. This is the error I get:

```
23-06-10 08:59:34 error: Error: connect EHOSTUNREACH 192.168.1.100:5000
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16)
```

With this config:

```
frigate:
  url: http://192.168.1.100:5000
```

**Reply by NickM-27** [reply]
No, I have over 50 images for each person. I just mean as I was in the training process I would go outside and walk around a bunch and then pick 2 images to add to the training set and then refresh all the matches to see if the score improved or not. If they got worse generally then I would remove those 2 images from the training set and pick different ones. Rinse and repeat

**Reply by DrSpaldo** [reply]
Ah ok, interesting, so you may have started with 60 images and ended up with 50 as some wouldn't help the score. I may have to do that with my kids as they are similar

**Reply by NickM-27** [reply]
I also started with pictures taken with a smart phone and cropped them, only adding photos from the camera itself after that

**Comment by W-Floyd** [comment]
Implemented in #14495
